process/sync/storageBootstrap/shardStorageBootstrapper.go

```diff
@@ -139,6 +140,80 @@ func (ssb *shardStorageBootstrapper) cleanupNotarizedStorage(shardHeaderHash []b
 	}
 }
 
+func (ssb *shardStorageBootstrapper) cleanupNotarizedStorageForHigherNoncesIfExist(
+	crossNotarizedHeaders []bootstrapStorage.BootstrapHeaderInfo,
+) {
+	var numConsecutiveNoncesNotFound int
+
+	nonce, err := getCrossNotarizedHeaderNonce(crossNotarizedHeaders)
+	if err != nil {
+		log.Warn("cleanupNotarizedStorageForHigherNoncesIfExist", "error", err.Error())
+		return
+	}
+
+	log.Debug("cleanup notarized storage higher than nonce", "nonce", nonce)
+
+	for {
+		nonce++
+
+		metaBlock, metaBlockHash, err := process.GetMetaHeaderFromStorageWithNonce(
+			nonce,
+			ssb.store,
+			ssb.uint64Converter,
+			ssb.marshalizer,
+		)
+		if err != nil {
+			log.Debug("meta block is not found in MetaHdrNonceHashDataUnit storage",
+				"nonce", nonce)
+
+			numConsecutiveNoncesNotFound++
+			if numConsecutiveNoncesNotFound > maxNumOfConsecutiveNoncesNotFoundAccepted {
+				break
```

Comment:
add a log indicating that the number of nonces to check has been exceeded

Reply:
Done